### PR TITLE
[BUG] store resolved rng_key as public attribute in BaseInferenceEngine

### LINKED ISSUE #357: [BUG] BaseInferenceEngine rng_key not preserved during skbase clone, causing non-determinism across kernel restarts
## Description

`MAPInferenceEngine` produces different results across kernel restarts even with identical data and parameters. In some sessions the optimizer diverges completely (WAPE > 40, negative predictions).

## Root cause

In `BaseInferenceEngine.__init__`, `rng_key` is stored as `None` (the default value) in the public attribute, while the resolved `PRNGKey` is only kept in the private `_rng_key`:

```python
def __init__(self, rng_key=None):
    self.rng_key = rng_key        # stores None
    if rng_key is None:
        rng_key = jax.random.PRNGKey(0)
    self._rng_key = rng_key       # stores the resolved key
```
When skbase clones the estimator internally during fit(), it calls get_params() which returns {"rng_key": None}, then reconstructs the engine with __init__(rng_key=None). This creates a new PRNGKey at an unpredictable point in JAX's lifecycle, generating different random sequences across restarts — sometimes causing the Adam optimizer to diverge over 20k steps.

## Proposed fix
Resolve rng_key before assigning to the public attribute, so get_params() returns the actual key and survives the skbase clone cycle:

```python
def __init__(self, rng_key=None):
    if rng_key is None:
        rng_key = jax.random.PRNGKey(42)
    self.rng_key = rng_key        # stores the resolved key
    self._rng_key = rng_key
```
Happy to open a PR if this looks good to you.

### PR DESCRIPTION
**Pull request by semissatto2**
Closes #357

## What changed

In `BaseInferenceEngine.__init__`, resolve `rng_key` before assigning to the public attribute.

## Why

When `rng_key=None` was stored as the public attribute, `get_params()` returned `None`. skbase clones the estimator internally during `fit()`, reconstructing the engine from `get_params()` — which re-called `__init__(rng_key=None)` and created a new `PRNGKey` at an unpredictable point in JAX's lifecycle. This caused non-determinism across kernel restarts, and in some cases complete optimizer divergence.

## Fix

```python
# Before
def __init__(self, rng_key=None):
    self.rng_key = rng_key        # stored None — lost on skbase clone
    if rng_key is None:
        rng_key = jax.random.PRNGKey(0)
    self._rng_key = rng_key

# After
def __init__(self, rng_key=None):
    if rng_key is None:
        rng_key = jax.random.PRNGKey(42)
    self.rng_key = rng_key        # stores resolved key — survives clone
    self._rng_key = rng_key

### CODE REVIEW

File: src/prophetverse/engine/base.py [hunk 1]
```diff
@@ -32,10 +32,9 @@ class BaseInferenceEngine(BaseObject):
     }
 
     def __init__(self, rng_key=None):
-        self.rng_key = rng_key
-
         if rng_key is None:
-            rng_key = jax.random.PRNGKey(0)
+            rng_key = jax.random.PRNGKey(42)
```
Comment (felipeangelimvieira) [inline src/prophetverse/engine/base.py]: Could we keep seed as 0 to avoid unexpected changes for models that already execute with Python hash seed fixed?

**Review by felipeangelimvieira — CHANGES_REQUESTED** [review]

File: src/prophetverse/engine/base.py (hunk 1 above)
Comment (semissatto2) [reply src/prophetverse/engine/base.py]: Done
